server/src/main/java/io/druid/guice/AWSModule.java

```diff
@@ -47,8 +50,17 @@ public AWSCredentialsProvider getAWSCredentialsProvider(final AWSCredentialsConf
 
   @Provides
   @LazySingleton
-  public AmazonEC2 getEc2Client(AWSCredentialsProvider credentials)
+  public AmazonEC2 getEc2Client(
+      AWSCredentialsProvider credentials,
+      ResourceManagementSchedulerConfig resourceManagementSchedulerConfig)
```

Comment:
we cannot make the AWS module dependent on indexing service classes, it should be generic enough that other druid nodes can use it as well. Not every druid node will have a proper ResourceManagementSchedulerConfig available for injection.